File: 32把数组排成最小的数/Solution.cpp
class Solution {
public:
    string PrintMinNumber(vector<int> numbers) {
        int len =numbers.size();
        string res;
        if(len==0) return "";
        sort(numbers.begin(),numbers.end(),cmp);
        for (int i=0;i<len;i++){
            res+=to_string(numbers[i]);
         }
        return res;
    }
private:
    static bool cmp(int a,int b){
        string ab=to_string (a)+to_string (b);
        string ba=to_string (b)+to_string (a);
        return ab<ba;
        
    }
};
/* 定义一个排序规则函数，将该函数作为参数传入主函数的sort中
逐个比较即可 */

File: 32把数组排成最小的数/Solution.java
import java.util.ArrayList;

public class Solution {
    public String PrintMinNumber(int [] numbers) {
  if (numbers == null || numbers.length == 0)
        return "";
    int n = numbers.length;
    String[] nums = new String[n];
    for (int i = 0; i < n; i++)
        nums[i] = numbers[i] + "";
    Arrays.sort(nums, (s1, s2) -> (s1 + s2).compareTo(s2 + s1));
    String ret = "";
    for (String str : nums)
        ret += str;
    return ret;
    }
}
/* java 的写法和cpp的区别在于，java需要先将数组转入字符串数组，但好处是
字符串的comparetoTo函数默认支持按字典序的元素排序 */